### PR TITLE
Record localization

### PR DESCRIPTION
**Pull request by h3ndrk**
## Introduced Changes

Records localization inputs and outputs in log file on the robot. Will automatically downloaded by Pepsi.

## ToDo / Known Issues

None

## Ideas for Next Iterations (Not This PR)

- Configure path in `hardware.json`

## How to Test

Deploy on a NAO, activate it in the parameters, run over the field, download the logs and observe that there is a file written. We currently don't have a viewer/reader for the file.

### CODE REVIEW

**Comment by schmidma** [comment]
I think, in the long run, we should integrate such a recording into our framework.
But I agree, that in the meantime, we can easily provide a quick solution with a recording node.
Still, I think this is a general topic and we should extend this node to support recording any dependency, not only localization. In the end this may boil down to only change naming from `localization*` to `*`.
What do you think?

File: crates/control/src/localization_recorder.rs [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+use bincode::serialize;
+use color_eyre::{eyre::Context, Result};
+use context_attribute::context;
+use framework::{HistoricInput, PerceptionInput};
+use nalgebra::Isometry2;
+use serde::{Deserialize, Serialize};
+use types::{GameControllerState, LineData, PrimaryState};
+
+pub struct LocalizationRecorder {
+    recording: Option<File>,
```
Comment (schmidma) [inline crates/control/src/localization_recorder.rs:17]: We should use a buffered writer here. e.g. `std::io::BufWriter`

Comment (h3ndrk) [reply crates/control/src/localization_recorder.rs:17]: Done

**Comment by h3ndrk** [comment]
> I think, in the long run, we should integrate such a recording into our framework. But I agree, that in the meantime, we can easily provide a quick solution with a recording node.

Exactly that was my reasoning. We had a lot of problems with localization on the challanging field and I don't know why, therefore the recording. And since the framework does not support that currently, it is implemented in the robotics domain.

> Still, I think this is a general topic and we should extend this node to support recording any dependency, not only localization. In the end this may boil down to only change naming from `localization*` to `*`. What do you think?

I hope, if we need to record more data on RoboCup 2023, we can extend this node by changing the code manually. After RoboCup, we should implement that in the framework properly. We should have a design session before that to align the requirements before implementation.

I didn't get your naming suggestion... :thinking: :sweat_smile:

**Review by knoellle — APPROVED** [review]
LGTM

**Pull request merged**